int32 row
int32 col
int32 status
---
int32 status
